packages/mobile/src/navigator/DrawerNavigator.tsx

```diff
@@ -0,0 +1,196 @@
+import ContactCircle from '@celo/react-components/components/ContactCircle'
+import PhoneNumberWithFlag from '@celo/react-components/components/PhoneNumberWithFlag'
+import colorsV2 from '@celo/react-components/styles/colors.v2'
+import fontStyles from '@celo/react-components/styles/fonts.v2'
+import {
+  createDrawerNavigator,
+  DrawerContentComponentProps,
+  DrawerContentOptions,
+  DrawerContentScrollView,
+  DrawerItemList,
+} from '@react-navigation/drawer'
+import BigNumber from 'bignumber.js'
+import * as React from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import deviceInfoModule from 'react-native-device-info'
+import Account from 'src/account/Account'
+import FiatExchange from 'src/account/FiatExchange'
+import { default as InviteScreen } from 'src/account/Invite'
+import {
+  defaultCountryCodeSelector,
+  e164NumberSelector,
+  nameSelector,
+  userContactDetailsSelector,
+} from 'src/account/selectors'
+import Support from 'src/account/Support'
+import BackupIntroduction from 'src/backup/BackupIntroduction'
+import ExchangeHomeScreen from 'src/exchange/ExchangeHomeScreen'
+import WalletHome from 'src/home/WalletHome'
+import { Namespaces } from 'src/i18n'
+import AccountKey from 'src/icons/navigator/AccountKey'
+import AddWithdraw from 'src/icons/navigator/AddWithdraw'
+import Gold from 'src/icons/navigator/Gold'
+import Help from 'src/icons/navigator/Help'
+import Home from 'src/icons/navigator/Home'
+import Invite from 'src/icons/navigator/Invite'
+import Settings from 'src/icons/navigator/Settings'
+import { useDollarsToLocalAmount, useLocalCurrencySymbol } from 'src/localCurrency/hooks'
+import { Screens } from 'src/navigator/Screens'
+import useSelector from 'src/redux/useSelector'
+import { stableTokenBalanceSelector } from 'src/stableToken/reducer'
+import { currentAccountSelector } from 'src/web3/selectors'
+
+const Drawer = createDrawerNavigator()
+
+function CustomDrawerContent(props: DrawerContentComponentProps<DrawerContentOptions>) {
+  const displayName = useSelector(nameSelector)
+  const e164PhoneNumber = useSelector(e164NumberSelector)
+  const contactDetails = useSelector(userContactDetailsSelector)
+  const defaultCountryCode = useSelector(defaultCountryCodeSelector)
+  const balance = useSelector(stableTokenBalanceSelector)
+  const bigNumBalance = balance ? new BigNumber(balance) : undefined
+  const localBalance = useDollarsToLocalAmount(balance)
+  const symbol = useLocalCurrencySymbol()
+  const { t } = useTranslation(Namespaces.global)
+  const account = useSelector(currentAccountSelector)
+  const appVersion = deviceInfoModule.getVersion()
+
+  return (
+    <DrawerContentScrollView {...props}>
+      <View style={styles.drawerTop}>
+        <ContactCircle thumbnailPath={contactDetails.thumbnailPath} size={64} />
+        <Text style={[fontStyles.large600, styles.nameLabel]}>{displayName}</Text>
```

Comment:
Nit: Can you fold the styles into `styles.nameLabel` via destructuring to avoid these style arrays. Not a big deal either way.